### PR TITLE
Improvements to Committee Assignments for multiple key requests

### PR DESCRIPTION
**Pull request by prestonvanloon**
Still needs a bit of work, but results are very promising. 

```
bazel test //beacon-chain/rpc/validator:go_default_test --test_arg=-test.v --test_output=streamed --test_arg=-test.bench=BenchmarkCommitteeAssignment --test_filter=Benchmark --test_arg=-test.benchmem  --test_arg=-test.cpuprofile=/tmp/cpu.profile --define=ssz=mainnet
```

At 16k validators:
```
goos: linux
goarch: amd64
BenchmarkCommitteeAssignment-8                 8         182708160 ns/op        85009310 B/op     952701 allocs/op
PASS
```
Previously was over 4 minutes for 8k validators. 

Test benchmark set up takes almost 3 minutes for the benchmark, maybe @0xKiwi has a way to load a mainnet state from disk. 

I also removed the span for validator index. This method almost always is a cache hit and can generate a lot of spans at scale. It serves little benefit and was a red flame on the flame graph.

### CODE REVIEW

**Comment by prestonvanloon** [comment]
This is complete. I'll add more tests tomorrow for sanity, then we can review and push forward.

**Comment by prestonvanloon** [comment]
Added feature flag.

**Comment by rauljordan** [comment]
Very nice stuff, can't wait - thanks for adding all the safety rails to this method.

File: beacon-chain/core/helpers/committee.go [hunk 1]
```diff
@@ -194,9 +194,79 @@ func AttestingIndices(bf bitfield.Bitfield, committee []uint64) ([]uint64, error
...
+//
+// 1. Determine the proposer validator index for each slot.
+// 2. Compute all committees.
+// 3. Determine the attesting slot for each committee.
+// 4. Construct a map of validator indices pointing to the respective committees.
+func CommitteeAssignments(state *pb.BeaconState, epoch uint64) (map[uint64]*CommitteeAssignmentContainer, map[uint64]uint64, error) {
+	if epoch > NextEpoch(state) {
+		return nil, nil, fmt.Errorf(
+			"epoch %d can't be greater than next epoch %d",
+			epoch, NextEpoch(state))
```
Comment (0xKiwi) [inline beacon-chain/core/helpers/committee.go]:
```suggestion
			epoch, 
			NextEpoch(state),
		)
```

**Pull request merged**